.github/workflows/yocto-build-deploy.yml

```diff
@@ -828,6 +828,20 @@ jobs:
             exit 1
           fi
 
+      - name: Verify kernel module signing key
+        if: inputs.sign-image == true
+        env:
+          SIGN_KMOD_KEY_APPEND: ${{ secrets.SIGN_KMOD_KEY_APPEND }}
+        run: |
+          if [ -z "${SIGN_KMOD_KEY_APPEND}" ]; then
+            echo "SIGN_KMOD_KEY_APPEND is not set, skipping verification"
+            exit 0
+          fi
+          ./balena-yocto-scripts/build/verify-kernel-keys.sh \
+            -d "${MACHINE}" \
+            -s "${SHARED_BUILD_DIR}" \
+            -c "${SIGN_KMOD_KEY_APPEND}"
```

Comment:
As a general rule we don't want to pass secrets to scripts outside the workflow (we made an exception for the build step so it would be identical to the developer flow) but in this case I think it's probably okay because SIGN_KMOD_KEY_APPEND isn't a high-risk secret. It's only treated as one because it may contain customer names.

Reply:
oh ok that's why you rather have it in the workflow inline!

I tried to run put that directly in the workflow but it was quite ugly.

We have to set up everything for running in Docker + the cert comparison.

But let's do that inline then :disappointed: